### ISSUE TITLE
SIA-R83 (Text is clipped) doesn't detect invisibility due to clip-path 

### ISSUE THREAD

**Issue by dan-tripp-siteimprove** [Siteimprove/alfa#1548]
The CSS properties clip-path and/or -webkit-clip-path seems to be cause a false positive for this rule.   With those CSS properties one clip it completely so that it's invisible, even while its bounding rect's width/height are much greater than zero.  If the text isn't visible then it doesn't meet the applicability for the rule so it shouldn't be checked.  But it is checked, and it fails. 

The HTML that I saw on a customer's page which exhibited this looked roughly like this:
```
<div style="clip-path: polygon(0px 0px,0px 0px,0px 0px); -webkit-clip-path: polygon(0px 0px,0px 0px,0px 0px); overflow: hidden; height: 10px; width: 10px;">lorem ipsum dolor</div>
```
But that HTML on its own doesn't reproduce the problem, so there's something I don't understand here.   @Jym77 I won't post the customer's page here but if you would like me to file a jira too or do anything else to be more specific on this, let me know.

**Comment by Jym77** [comment]
I think we need to gather cases of how `clip-path` is used to make content invisible.
I do not want to implement a generic area computation of all possible `clip-path`, but I assume that there are only a handful that are commonly used to make content invisible, and we can likely make a heuristic to detect these cases specifically.

**Comment by dan-tripp-siteimprove** [comment]
Would you like any more info on this current case?

**Comment by Jym77** [comment]
> Would you like any more info on this current case?

No, I think there is enough with that CSS value. I'd like to know if it's somewhat common to use `polygon(0px, …)` and worth having a "`polygon` with all 0 is invisible" heuristic.

**Comment by wittjeff** [comment]
I dug into this and can explain both the false positive and why the minimal snippet doesn't reproduce it.

**Root cause.** `isVisible` → `isClipped` → `isClippedByMasking` only consults the deprecated `clip` property. `clip-path` is parsed by `alfa-style` but never used by any visibility predicate, so content fully masked by it is still considered visible and enters R83's applicability.

**Why the snippet doesn't reproduce.** `hasFixedDimension` in the R83 expectation logic deliberately ignores dimensions set via the `style` attribute (they're assumed to be JS-controlled and to scale with content). With every declaration inline, no constraining ancestor is found, the text is considered to have room to grow, and the snippet **passes**. Move the same declarations into a stylesheet — as presumably on the customer page — and it **fails**:

| variant | outcome |
|---|---|
| issue's HTML, all inline | passed |
| same CSS in a stylesheet | **failed** ← the false positive |
| inline + `white-space: nowrap` | failed |
| stylesheet, without any `clip-path` | failed (identical, i.e. `clip-path` is ignored) |
| `clip: rect(0,0,0,0)` + `position: absolute` | inapplicable (correctly handled) |
| `visibility: hidden` | inapplicable (correctly handled) |

(Verified on 0.112.0 and on current `main`.)

**A second, compounding bug.** While testing I found that `Polygon.parse` rejects every polygon accepted by browsers: the vertex list is parsed as whitespace-separated, but [css-shapes-1](https://drafts.csswg.org/css-shapes/#funcdef-basic-shape-polygon) defines it as a comma-separated list of coordinate pairs. `polygon(0px 0px, 0px 0px, 0px 0px)` → `Unexpected token ,` → `clip-path` computes to `none`. So even a `clip-path`-aware `isVisible` would never see this shape. (The existing polygon spec tests encode the same whitespace grammar.)

**On "which values are commonly used to hide":** the idioms I've seen in the wild are the all-zero / single-point `polygon()`, `inset(50%)` / `inset(100%)`, and `circle(0)`. All are covered by a "shape is *guaranteed* empty" heuristic — no general area computation needed, and misses can only create false negatives.

**Comment by wittjeff** [comment]
I've implemented the above as two commits — comma-separated `polygon()` parsing in `alfa-css`, and empty-shape detection in `isClippedByMasking` in `alfa-style` — with spec tests for both, an R83 regression test using this issue's exact shape (verified to fail before the fix), and changesets. The full suite passes (345 files / 3655 tests).

I'm aware R83 has since been deprecated (#2079), but the `clip-path` blindness sits in `isVisible`, which many other rules use for applicability, so the fix should still be worthwhile beyond R83.

The change is staged on my fork for now: https://github.com/wittjeff/alfa/pull/1 — happy to open it here (as-is, or reworked to your preferences) if you'd take an external PR for it.

**Comment by Jym77** [comment]
Thanks, the PR (for improved visibility) is ready, I hope we can release the new version next week.
